taichi/program/program.cpp

```diff
@@ -158,7 +158,6 @@ Program::Program(Arch desired_arch) : snode_rw_accessors_bank_(this) {
   SNode::counter = 0;
 
   result_buffer = nullptr;
-  sync = true;
```

Comment:
what is this for originally?

Reply:
For AsyncTaichi? The #5523 removed it. @lin-hitonami , WDYT?

Reply:
Yea IIRC this was legacy from async taichi.